### ISSUE TITLE
Search against index or index/type yields different results with Index containing Single Type

### ISSUE THREAD

**Issue by agastya71** [elastic/elasticsearch#24019]
**Elasticsearch version**: 2.3.3

**Plugins installed**: [kopf]

**JVM version**:1.8.0_45

**OS version**: Oracle Linux 7.0

**Description of the problem including expected versus actual behavior**: Document's score is not calculated consistently and causes inconsistent results in Version 2.3.3, this behavior is not observed in ES 2.1.1.  This problem is exposed in a couple of different ways.  Executing a GET on the index/_search endpoint returns inconsistent results. Its cyclical (depending on the data),  every nth try might give the right result.   Another behavior that is noticed is, searching against an Index with a Single Type   with end point index/_search yields different results that with end point  index/type/_search. In theory this should not happen on indices with a single type. 

**Steps to reproduce**:  Cluster configuration : 3 ES nodes performing all 3 roles of master, data and client nodes).  Index with a Single shard  with replication of 2. Index content with a mapping containing a Multi-field definition, one of the fields in the multi-field having a Custom edge ngram analyzer, one with  field not analyzed and one with the default analyzer (field is of type text/String)
 1. in an ES 2.3.3 instance, on a static index with Millions of Documents, execute  GET index/_search notice results if it changes with each subsequent execution. Also notice if the results cycle.
 2. in the same ES 2.3.3 instance, execute a search  against end points index/_search and index/type/_search  Notice if there is a difference. 
 3.

**Provide logs (if relevant)**:  link to forum post on same issue : 

1. https://discuss.elastic.co/t/difference-between-index-type--search-and-index--search-results/81314  
2. https://discuss.elastic.co/t/multi-match-query-with-filter-condition-differences-between-es2-1-1-and-es-2-3-3/80732/4

**Comment by spinscale** [comment]
please keep the discussion in the forum and provide information there (no mappings/index creation examples to replay were provided, no exact queries, no mention if using a preference changed behaviour, screenshots instead of gists make it really hard to follow the problem - if the reporter was you, otherwise opening a new discuss post makes most sense IMO)
